components/date_picker/date_picker.tsx

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useRef, useState} from 'react';
+
+import FocusTrap from 'focus-trap-react';
+import {DayPicker, DayPickerProps} from 'react-day-picker';
+import {usePopper} from 'react-popper';
+
+import type {Locale} from 'date-fns';
+
+import 'react-day-picker/dist/style.css';
+import './date_picker.scss';
+
+type Props = {
+    children: React.ReactNode;
+    datePickerProps: DayPickerProps;
+    isPopperOpen: boolean;
+    locale: string;
+    closePopper: () => void;
+}
+
+const DatePicker = ({children, datePickerProps, isPopperOpen, closePopper, locale}: Props) => {
+    const loadedLocales: Record<string, Locale> = {};
```

Comment:
Making this a variable inside the functional component means that it won't be persisted between runs of this, so it'll have to be a global variable like the one in `suggestion_date_picker`. Since we're doing the same thing in both places, perhaps it would be better to pull this out into a reusable utility function of some sort? I'd say a hook, but I don't want to send us down the rabbit hole of trying to make the `Suggestion` components functional right now 😛

Reply:
Moved to a utility function in utils.tsx